pkg/resource/proxy-auth/proxy-auth.go

```diff
@@ -0,0 +1,118 @@
+package proxyauth
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+
+	"github.com/giantswarm/logging-operator/pkg/common"
+	loggedcluster "github.com/giantswarm/logging-operator/pkg/logged-cluster"
+	loggingcredentials "github.com/giantswarm/logging-operator/pkg/resource/logging-credentials"
+)
+
+const (
+	//#nosec G101
+	proxyauthSecretName      = "grafana-multi-tenant-proxy-auth-config"
+	proxyauthSecretNamespace = "monitoring"
+	// DefaultReadOrgIDs - make sure to have at least 2 tenants, to prevent writing with this user
+	DefaultReadOrgIDs = "giantswarm|default"
+)
+
+type Values struct {
+	Users []user `yaml:"users" json:"users"`
+}
+
+type user struct {
+	Username string `yaml:"username" json:"username"`
+	Password string `yaml:"password" json:"password"`
+	Orgid    string `yaml:"orgid" json:"orgid"`
+}
+
+// ProxyConfigSecretMeta returns metadata for the grafana-multi-tenant-proxy secret
+func ProxyAuthSecretMeta(lc loggedcluster.Interface) metav1.ObjectMeta {
+	metadata := metav1.ObjectMeta{
+		Name:      proxyauthSecretName,
+		Namespace: proxyauthSecretNamespace,
+		Labels:    map[string]string{},
+	}
+
+	common.AddCommonLabels(metadata.Labels)
+	return metadata
+}
+
+// listWriteUsers returns a map of users found in a credentialsSecret
+func listWriteUsers(credentialsSecret *v1.Secret) []string {
+	var usersList []string
+	for myUser := range credentialsSecret.Data {
+		// bypass read user
+		// bypass old creds (xxxuser and xxxpassword)
+
+		if !strings.HasSuffix(myUser, "user") && !strings.HasSuffix(myUser, "password") && myUser != common.ReadUser {
+			usersList = append(usersList, myUser)
+		}
+	}
+
+	return usersList
+}
+
+// GenerateProxyAuthSecret returns a secret for
+// the grafana-multi-tenant-proxy auth config
+func GenerateProxyAuthSecret(lc loggedcluster.Interface, credentialsSecret *v1.Secret) (v1.Secret, error) {
+	// Init empty users structure
+	values := Values{
+		Users: []user{},
+	}
+	// Prepare read user's orgid with default values
+	readOrgid := DefaultReadOrgIDs
+
+	// Loop on write users
+	for _, writeUser := range listWriteUsers(credentialsSecret) {
+
+		writePassword, err := loggingcredentials.GetPassword(lc, credentialsSecret, writeUser)
+		if err != nil {
+			return v1.Secret{}, errors.WithStack(err)
+		}
+
+		values.Users = append(values.Users, user{
+			Username: writeUser,
+			Password: writePassword,
+			// we set the tenant even though it may be given by the sender (promtail)
+			// depending of grafana-multi-teant-proxy config
+			Orgid: writeUser,
```

Comment:
How does this work with keep org id setting ?

Reply:
You know I copied this from loki-auth to recreate the secret, I was not trying to change the implementation :D

Reply:
Well this is the enforced value for the user

Reply:
Yes